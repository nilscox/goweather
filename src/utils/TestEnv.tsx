import * as React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter as Router } from 'react-router-dom';

import store from '../store';

type TestEnvProps = {
  children: React.ReactNode;
};

const TestEnv = ({ children }: TestEnvProps) => (
  <Provider store={store}>
    <Router>
      { children }
    </Router>
  </Provider>
);

export default TestEnv;
