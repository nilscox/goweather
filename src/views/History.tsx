/** @jsx jsx */
import { jsx } from '@emotion/core';
import { Dispatch } from 'redux';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { Button, Container } from 'reactstrap';

import { clearHistory } from '../store/actions';
import { State } from '../store/state';
import { ICity } from '../interfaces';
import { PageTitle } from '../components';

type HistoryProps = {
  cities: ICity[];
  clearHistory: () => any;
};

const mapStateToProps = (state: State) => ({
  cities: state.history,
})

const mapDispatchToProps = (dispatch: Dispatch) => ({
  clearHistory: () => dispatch(clearHistory()),
})

/**
 * History page. Displays a list of cities that have already been searched.
 */
const History = ({ cities, clearHistory }: HistoryProps) => (
  <Container className="pb-4">

    <PageTitle>Search history</PageTitle>

    <Container>
      { cities.map((city: ICity) => (
        <Link
          key={city.id}
          to={'/forecast/' + city.id}
          className="d-block py-2 px-4 m-2 bg-light"
        >
          { city.name } ({ city.country })
        </Link>
      )) }

      <Button onClick={clearHistory}>Clear</Button>

    </Container>

  </Container>
);

export default connect(mapStateToProps, mapDispatchToProps)(History);
