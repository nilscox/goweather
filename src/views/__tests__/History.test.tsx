import * as React from 'react';
import * as renderer from 'react-test-renderer';

import TestEnv from '../../utils/TestEnv';

import History from '../History';

test('History', () => {
  const tree = renderer
    .create(<TestEnv><History /></TestEnv>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
