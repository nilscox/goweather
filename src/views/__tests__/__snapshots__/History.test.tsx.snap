// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`History 1`] = `
<div
  className="pb-4 container"
>
  <h1
    className="text-center py-4 css-1aylpg"
  >
    Search history
  </h1>
  <div
    className="container"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Clear
    </button>
  </div>
</div>
`;
